lib/vfscore/main.c

```diff
@@ -498,24 +537,44 @@ UK_TRACEPOINT(trace_vfs_pwrite, "%d %p 0x%x 0x%x", int, const void*, size_t,
 UK_TRACEPOINT(trace_vfs_pwrite_ret, "0x%x", ssize_t);
 UK_TRACEPOINT(trace_vfs_pwrite_err, "%d", int);
 
-ssize_t pwrite(int fd, const void *buf, size_t count, off_t offset)
+/*
+ * Some libc's define some macros that remove the 64 suffix
+ * from some system call function names (e.g., <unistd.h>, <fcntl.h>).
+ * We need to undefine them here so that our system call
+ * registration does not fail in such a case.
+ */
+#ifdef pwrite64
+#undef pwrite64
+#endif
+
+UK_LLSYSCALL_R_DEFINE(ssize_t, pwrite64, int, fd,
+		      const void *, buf, size_t, count, off_t, offset)
 {
 	trace_vfs_pwrite(fd, buf, count, offset);
 	struct iovec iov = {
 			.iov_base	= (void *)buf,
 			.iov_len	= count,
 	};
-	int bytes;
+	ssize_t bytes;
 
-	bytes = pwritev(fd, &iov, 1, offset);
+	bytes = uk_syscall_r_pwritev((long) fd, (long) &iov,
+				     (long) 1, (long) offset);
 	if (bytes < 0)
 		trace_vfs_pwrite_err(bytes);
 	else
 		trace_vfs_pwrite_ret(bytes);
 	return bytes;
 }
 
+#if UK_LIBC_SYSCALLS
+ssize_t pwrite(int fd, const void *buf, size_t count, off_t offset)
```

Comment:
This does bring the libc-style wrapper for `pwrite` but causes an undefined reference if the libc-style wrapper of `pwrite64` were to be used.

Reply:
This is the same as the `pread` argument. I will provide both `64` versions of the symbol.